src/site/apt/user_indexing.apt

```diff
@@ -0,0 +1,200 @@
+you should do the following.
+
+*** Configuring VXQuery to use indexing functions.
+Open local.xml file in your vxquery/vxquery-server/src/main/resources/conf directory. Then add the following line.
+
+--------
+<indexDirectory>path_to_index_directory</indexDirectory>
+--------
+(You should create the index_directory)
+
+** Using indexing queries.
+
+VXQuery offers following indexing functionality.
+
+[[a]] Create an index for collection.
+[[b]] Use the index in executing a query.
+[[c]] Update the index.
+[[d]] Delete the index.
+[[e]] View existing indexes.
+
+*1. Scenario I - When collection is a single directory.
+In this scenario, all the XML files are stored in a single directory. (There can be sub directories)
+
+*** Creating an index for collection
+If I need to create index for xml collection stored <path_to_collection>/collection1,
+
+Query structure:
+--------
+build-index-on-collection("<path_to_collection>")
+--------
+
+You can see the index has created in a new sub-directory in the index_directory specified in local.xml
+
+Example:
+--------
+build-index-on-collection("<path1>/collection1")
+--------
+This function takes the collection path as an argument.
+
+*** Using index in query.
+If we need to use the index and execute a query, use the following structure.
+
+------
+for $r in collection-from-index("<path1>/collection1", "/dataCollection/data")/data
+where $r/dataType eq "AWND" and xs:decimal($r/value) gt 491.744
+return $r
+------
+Here the index access function is,
+
+------
+collection-from-index
+------
+
+which takes two arguments, collection folder and the path element.
+
+Result
+
+------
+<data><date>2001-01-01T00:00:00.000</date><dataType>AWND</dataType><station>GHCND:US000000001</station><value>1000</value><attributes><attribute/><attribute/><attribute>a</attribute></attributes></data>
+------
+
+*** Updating the index.
+A collection can be modified or changed by following ways.
+[[1]] Inserting new XML files.
+[[2]] Deleting files.
+[[3]] Add/ remove or modify the content of XML files.
+
+In this type of situation, the index corresponding to the modified collection must also be modified. To achieve this
+the update-index function can be used.
+
+Query structure:
+--------
+update-index("<path_to_collection>")
+--------
+
+Example:
+-------
+update-index("<path1>/collection1")
+-------
+
+This function takes the collection which was modified.
+
+*** Deleting the index.
+If we want to delete the entire index created for a collection, the delete-index function can be used.
+This function also takes the collection path of which the index is needed to be deleted.
+
+Query structure:
+--------
+delete-index("<path_to_collection>")
+--------
+
+Example:
+-------
+delete-index("<path_to_collection>/collection1")
+-------
+
+*2. Scenario II - When the collection is distributed.
+In this scenario, the collection is distributed among several partitions.
```

Comment:
In this scenario, the collection is distributed among several directories. We can distribute the queries among partitions